chemprop/data/utils.py

```diff
@@ -202,6 +201,7 @@ def get_data(path: str,
         # By default, the SMILES column is the first column
         if None in smiles_columns:
             smiles_columns = columns[:len(smiles_columns)]
+            args.smiles_columns=smiles_columns
```

Comment:
Could there be any other effects from reassigning this value in args? E.g. if any downstream functions expect the original value.

Reply:
I can't find any places where the assignment would be a diversion from what's done already. In every instance where something calls `args.smiles_columns` or wants to use it, there is a logic check where
>if None in smiles_columns:
     smiles_columns = reader.fieldnames[:len(smiles_columns)]

This is the same as the check that's done before the assignment of `args.smiles_columns` that I added, so the effect everywhere that it's used would be the same.

The only place where the code calls for `args.smiles_columns` as keys for a dict that could be disrupted are in the `molecule.py` file that I am adding, which specifically just works because I added that new assignment.

The thing that I could see changing in the code at large is centralizing this check and doing the reassignment of `args.smiles_columns` really early . Right now it's done redundantly all over the place. But I could change it so that this is done in the initial args parsing and then remove all of these equivalent redundant checks. And by doing it early, that would preclude any conflicting inconsistent references (that don't seem to exist now but could be accidentally written in the future).